pkg/bgpv1/manager/reconcilerv2/policies.go

```diff
@@ -197,6 +197,106 @@ func CreatePolicy(name string, peerAddr netip.Addr, v4Prefixes, v6Prefixes types
 	return policy, nil
 }
 
+// MergeRoutePolicies evaluates two instances of RoutePolicy{} and returns a single RoutePolicy{} representing
+// the merger of the two.  The merge operation focuses on each policy's Statements.  Statements define one or more
+// Actions, and these actions may include setting BGP Communities, Local Preference, and others.  Statements are
+// keyed by their Conditions, which define the BGP Neighbor, AF, and Prefixes it applies to.
+//
+// The merge operation evaluates Actions across only those statements with the same key. For these Statements,
+// the merge takes the union of all BGP Communities set.  When differing Local Preference values are set, the
+// higher value is selected.
+func MergeRoutePolicies(policyA *types.RoutePolicy, policyB *types.RoutePolicy) (*types.RoutePolicy, error) {
+	if policyA == nil || policyB == nil {
+		return nil, fmt.Errorf("route policy is nil")
+	}
+	if policyA.Name != policyB.Name {
+		return nil, fmt.Errorf("route policy names do not match")
+	}
+	if policyA.Type != policyB.Type {
+		return nil, fmt.Errorf("route policy types do not match")
+	}
+
+	mergedPolicy := &types.RoutePolicy{
+		Name:       policyA.Name,
+		Type:       policyA.Type,
+		Statements: []*types.RoutePolicyStatement{},
+	}
+
+	// Maps a string key representing a unique RoutePolicyConditions{} (where unique is defined by the values of all
+	// attributes) to a RoutePolicyStatement{}.  When multiple instances of the same key representing
+	// RoutePolicyConditions{} are observed, the existing RoutePolicyStatement{} will be updated to reflect the union
+	// of all instances.
+	conditionsToMergedRoutePolicyStatement := map[string]*types.RoutePolicyStatement{}
+
+	// Extract the first policy's statements and communities
+	for _, statement := range policyA.Statements {
```

Comment:
As you look through this loop, and the duplicate of it starting on line 257 for `policyB`, I am curious to hear if you would replace both with a function call?

I am leaning that way, but it would add the complexity related to passing around and updating the map.

Reply:
Yep, let's replace this with function call. I suspect this code will grow in future as we add support for more attributes.